WordPress/openverse-frontend
# Set maximum height to the popover to prevent cut-off on smaller screens

## Fixes


Fixes #1212 by @Asadpalasara-multidots

Also fixes #960

## Description


This PR adds a `clippable` prop to `VPopover` that makes the popover only take up the available window height and scroll if the content does not fit.
Currently, when the popover content becomes scrollable, I can also see a scroll line in the bottom. I need to figure out how to remove it.

## Testing Instructions

Open a search results page, make the window height small so that the menu does not fit in. Then, click on the page menu or the content switcher. You should see that the popover is only as high as the window, and its content is scrollable:

![clip-popover](https://user-images.githubusercontent.com/15233243/185585961-413f5d9c-230c-40bb-9ea4-34402165b5e6.gif)

The popover should also update if you change the window height.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
